dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Current/PhaseShifterBlockingI.mo

```diff
@@ -0,0 +1,47 @@
+within Dynawo.Electrical.Controls.Current;
+
+/*
+* Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model PhaseShifterBlockingI "Phase Shifter blocking model"
+  import Modelica.Constants;
+  import Dynawo.NonElectrical.Logs.Timeline;
+  import Dynawo.NonElectrical.Logs.TimelineKeys;
+  import Dynawo.Types;
+
+  public
+    parameter Types.CurrentModule IMax "Maximum current on the monitored component";
+    parameter Types.Time tLagBeforeActing "Time lag before taking action";
+
+    input Types.CurrentModule IMonitored "Monitored current";
+    output Boolean locked "Is phase shifter locked?";
```

Comment:
I don' t know what we decided regarding the use of the keywords "input" and "output", in particular outside regulations. Of course, there is a causality in this model but from a general manner, the problem we solve remains acausal so I'm not a big fan of using them. @cossartque any opinion?

Reply:
For me, we don't use them when in a physical model but we do use them in regulations. Here what is implemented is closer to a regulation than to a physical model (we measure the current and decide to lock depending on it, so there is an input/output relation here for me). I would keep the input/output here.

Reply:
Ok then. It's right that it is closer to a regulation than a physical model.